t/001-tls.t

```diff
@@ -538,3 +538,361 @@ ok
+MIIDlTCCAn2gAwIBAgIUEYBZNDoOJlmg1B3lCS0WCk8bd2gwDQYJKoZIhvcNAQEN
+BQAwWjEQMA4GA1UEAwwHeHh4LmNvbTELMAkGA1UEBhMCQ04xCzAJBgNVBAgMAkpT
+MRAwDgYDVQQHDAdKaWFuZ3N1MQwwCgYDVQQKDANBQUExDDAKBgNVBAsMA0JCQjAe
+Fw0yNTA4MDQxMjUyNDRaFw0zNTA4MDIxMjUyNDRaMFoxEDAOBgNVBAMMB3h4eC5j
+b20xCzAJBgNVBAYTAkNOMQswCQYDVQQIDAJKUzEQMA4GA1UEBwwHSmlhbmdzdTEM
+MAoGA1UECgwDQUFBMQwwCgYDVQQLDANCQkIwggEiMA0GCSqGSIb3DQEBAQUAA4IB
+DwAwggEKAoIBAQDTuNAcos+LA//fjl1qr3lUOAIQczp9wN3hoQ1v/Yt9m+4rLJcu
+dewGT0o+tOMXAHYVo3KyRfVTdpGNUAZBOpLC5x20LYhzGOM9vL+VnZ/jci7poPsF
+ynNWLd/JGUOlv68JHyNFG+ghIgEym/Lu4qAC5REvyO7D988zzvOEY9gTV33VRRph
++OI//eSqEfGKj4s2Yfg9aXMD2gK+ORtJx1bMo98p35plyYQWm3kQWq/pUm+7LXbR
+q7Zp7I611G0XDdSVAt/+SzhNUKOxyMSrpGUdlRxQsR7OhwITiOxN4Pp4lhlQRHFA
+T8sQum/+SrVeXtAdZnJEn0Aj2y+P70+vhjUTAgMBAAGjUzBRMB0GA1UdDgQWBBTn
+Jn6CXjeLpRXdqpCUqofdgDMh8zAfBgNVHSMEGDAWgBTnJn6CXjeLpRXdqpCUqofd
+gDMh8zAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBDQUAA4IBAQDJfIOc/uv0
+Q2Ob/EjXas8x1kLst9ktT8XiAYg1P8y2KZJGnJ/M0bEgyJGNdXJMfQjEntbjwLm9
+L0qdROfKb1WeKWfCXI49gNmErtddHUAhHLIlm9W8hCGE6yH7VsEfE/6e2L4qV6RO
+tWmGu5ZTAMi2mInJsFojq+q4IQAeXeEigde5i83TjRi9o56f7TcAcnTBhuXNPAuK
+ULzbPEqPUw5Au6EsW2Y9X3Vg/qRsMLEJBk+2QVaG11lOIYEVgW+LbX7HywGf6E43
++U4EWZfeqaMsiYgh1ah3H9JD6RIxoy6VaOV88lnGs/Qi5faP5Z4rIoDTo1wsfCwE
+MQVJX0HYQMLS
+-----END CERTIFICATE-----
+--- no_error_log
+[error]
+[emerg]
+
+
+
+
+=== TEST 13: Test ngx.var.upstream_ssl_protocol can get in log phase after accessed in header_filter phase with lua_kong_load_var_index set to default
+--- http_config
+    lua_package_path "../lua-resty-core/lib/?.lua;lualib/?.lua;;";
+    lua_ssl_protocols SSLV3 TLSv1 TLSv1.1 TLSv1.2;
+    lua_kong_load_var_index default;
+
+    # This is the upstream server
+    server {
+        listen unix:$TEST_NGINX_HTML_DIR/upstream.sock ssl;
+        server_name   upstream.example.com;
+        ssl_certificate ../../cert/upstream.crt;
+        ssl_certificate_key ../../cert/upstream.key;
+        ssl_session_cache off;
+        server_tokens off;
+
+        location / {
+            content_by_lua_block {
+                ngx.say("testtes") -- clear warning
+            }
+        }
+    }
+
+--- config
+    server_tokens off;
+    location /t {
+        proxy_pass https://unix:$TEST_NGINX_HTML_DIR/upstream.sock;
+        proxy_ssl_server_name on;
+        proxy_ssl_name upstream.example.com;
+        proxy_ssl_session_reuse off;
+        proxy_ssl_protocols TLSv1.2;
+
+        header_filter_by_lua_block {
+            local upstream_tls_version = ngx.var.upstream_ssl_protocol
+            if upstream_tls_version then
+                ngx.header["X-Upstream-Ssl"] = upstream_tls_version
+            else
+                ngx.header["X-Upstream-Ssl"] = "No upstream TLS version available"
+            end
+        }
+
+        log_by_lua_block {
+            local upstream_tls_version = ngx.var.upstream_ssl_protocol
+            if upstream_tls_version then
+                ngx.log(ngx.ERR, "Upstream TLS version: " .. upstream_tls_version)
```

Comment:
Let's use lower log level like INFO so that we can confirm there are no actual error level logs.

```
--- no_error_log
[error]
[crit]
[alert]
[emerg]
```